src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -728,6 +728,45 @@ public function readFile(string $filename): string
         return $content;
     }
 
+    /**
+     * Reads a file in fixed-size chunks and yields them one at a time.
+     *
+     * @param string $filename The full path to the file
+     *
+     * @return \Generator<string> Yields file content as strings in chunks
+     *
+     * @throws IOException If the file cannot be opened or read, or if it's a directory
+     */
+    public function readFileInChunks(string $filename, int $size = 8192): \Generator
```

Comment:
I think the return type should be either `iterable` or `\Traversable` (if we want to guarantee that it is not returning an array) rather than `\Generator`, as it gives us more flexibility to refactor the implementation in the future (widening a return type is a BC break)